### ISSUE TITLE
false negative of `vue/no-mutating-props` when mutating deconstructed prop

### ISSUE THREAD

**Issue by n0099** [vuejs/eslint-plugin-vue#2548]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 
- **eslint-plugin-vue version:** 
- **Vue version:** 
- **Node version:** 
- **Operating System:** 

**Please show your full configuration:**
https://github.com/n0099/open-tbm/blob/fec399a0c9e462c77aaf5e6d5a081529428d8597/fe/eslint.config.js

**What did you do?**	
https://github.com/n0099/open-tbm/blob/fec399a0c9e462c77aaf5e6d5a081529428d8597/fe/src/components/post/queryForm/QueryForm.vue#L241
```diff
-        uniqueParams.value.postTypes.value = from; // to prevent empty post types
+        props.queryFormDeps.uniqueParams.value.postTypes.value = from; // to prevent empty post types
```

**What did you expect to happen?**
Reporting.

**What actually happened?**
Not reporting.

**Repository to reproduce this issue**
https://github.com/n0099/open-tbm/blob/fec399a0c9e462c77aaf5e6d5a081529428d8597/fe
